sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -119,21 +146,33 @@ object GpuFilter extends Arm {
     }
   }
 
-  def apply(
-      batch: ColumnarBatch,
+  def apply(batch: ColumnarBatch,
       boundCondition: Expression) : ColumnarBatch = {
-    var filterConditionCv: GpuColumnVector = null
-    var tbl: cudf.Table = null
-    var filtered: cudf.Table = null
-    try {
-      filterConditionCv = boundCondition.columnarEval(batch).asInstanceOf[GpuColumnVector]
-      tbl = GpuColumnVector.from(batch)
-      val colTypes =
-        (0 until batch.numCols()).map(i => batch.column(i).dataType())
-      filtered = tbl.filter(filterConditionCv.getBase)
-      GpuColumnVector.from(filtered, colTypes.toArray)
-    } finally {
-      Seq(filtered, tbl, filterConditionCv, batch).safeClose()
+    withResource(batch) { batch =>
+      val checkedFilterMap = withResource(
+        GpuProjectExec.projectSingle(batch, boundCondition)) { filterMap =>
+        // If  filter is a noop then return a None
+        withResource(filterMap.getBase.all()) { allIncluded =>
```

Comment:
Has there been any performance analysis of adding this extra reduction operation and synchronization to check for the filter corner case?  Should be a nice memory win for the common case of filtering for nulls after loading from input with no nulls, but not sure how much this is costing for the other cases.

Reply:
I ran q24a and q95. I didn't see anything in the nsys profile that indicated that this was causing a performance change at all good or bad. I am happy to remove it if you feel the complexity is not worth it. 

The profile put the `all` check at about the same speed as calculating the null count for a column. But that said I also didn't see any significant decrease in the use of contiguous split either. There really are just not that many cases where the input to a filter is already contiguous. They typically are happening right after a join, in our case, or after reading in input. This is because the filters are pushed to be as soon as possible in the query.

The profiles for the project also showed very little change. It is just metadata, but I did see a small drop in the number of contig splits needed for shuffled joins, but it was only on the data side not the key side. I was going to file a follow on issue to look into it more, but like I said I am happy to remove the extra code if the complexity is not worth it.